File: Sorting and Searching/ThreeValues.java
import java.io.*;
import java.util.*;
public class ThreeValues{
    public static void main(String args[])throws IOException
    {
    Scanner sc=new Scanner(System.in);
    int n=sc.nextInt();
    long x=sc.nextLong();
    long arr[]=new long[n];
    long sum=0;
    boolean flag=false;
    for(int i=0;i<n;i++){
        arr[i]=sc.nextLong();
    }
    Arrays.sort(arr);
    for(int i=0;i<n-2;i++)
    {
        sum=x-arr[i];
        int l=i+1;
        int r=n-1;
        while(l<r)
        {
            if(arr[l]+arr[r]==sum)
            {
                System.out.println((i+1) +" "+(l+1)+" "+(r+1));
                flag=true;
            }
            else if(arr[l]+arr[r]<sum)
                l++;
            else
                r--;
        }
    }
    if(!flag)
        System.out.println("IMPOSSIBLE");


}
}

File: Sorting and Searching/ThreeValues.cpp
#include <bits/stdc++.h>
#define lli long long int
using namespace std;

int main()
{
    int n,x,y;
    cin>>n>>x;
    vector<pair<int,int>> arr;
    for(int i=0;i<n;i++){
        cin>>y;
        arr.push_back(make_pair(y,i+1));
    }
    sort(arr.begin(),arr.end());
    lli sum=0;
    for(int i=0;i<n-2;i++)
    {
        sum=x-arr[i].first;
        int l=i+1;
        int r=n-1;
        while(l<r){
        if(arr[l].first+arr[r].first==sum){
            cout<<arr[i].second<<" "<<arr[l].second<<" "<<arr[r].second;
            return 0;
        }
        else if(arr[l].first+arr[r].first>sum)
            r--;
        else
            l++;
    }
    }
    cout<<"IMPOSSIBLE";
    return 0;
}